cmd/minishift/cmd/util/oc.go

```diff
@@ -92,3 +92,20 @@ func SetOcContext(profileName string) error {
 
 	return nil
 }
+
+//RemoveCurrentContext removes the current context from `machinename_kubeconfig`
+func RemoveCurrentContext() error {
+	requestedOpenShiftVersion := viper.GetString(configCmd.OpenshiftVersion.Name)
```

Comment:
that does not make any sense. How can you reference _configCmd_ here? This value would only be set as part of the _start_ command. In all other cases you get the default/baseline version. 

So for example, if you start your first cluster against 3.5, we would cache _oc_ 3.5. When you then use profile set to switch to another profile, I would assume you are downloading 3.6 as well and use it to set the context. Don't we write the oc path into the instance config? That's where it should come from. If not, we might have to use the default/baseline _oc_, but at least this should be implicit. @LalatenduMohanty ^^

Reply:
`configCmd.OpenshiftVersion.Name` is only the referenced name for the configuration parameter.
`viper.GetString()` would read the configured or specified parameter by retrieving the name

for example:

```
$ minishift config set vm-driver virtualbox
$ minishift start
```

will do a pre-flight check with the virtualbox driver:
```
-- Checking if requested hypervisor 'virtualbox' is supported on this platform ... OK                                                      
...
-- Starting local OpenShift cluster using 'virtualbox' hypervisor ...
```

In code it does:
https://github.com/minishift/minishift/blob/a9fc054fca4820cb9e249b71e6c28b01a7b123f8/cmd/minishift/cmd/start_preflight.go#L53

So I am not completely following the comment about `configCmd` being an issue here.

Reply:
> that does not make any sense. How can you reference configCmd here? This value would only be set as part of the start command. In all other cases you get the default/baseline version.
> 
> So for example, if you start your first cluster against 3.5, we would cache oc 3.5. When you then use profile set to switch to another profile, I would assume you are downloading 3.6 as well and use it to set the context. Don't we write the oc path into the instance config? That's where it should come from. If not, we might have to use the default/baseline oc, but at least this should be implicit

@hferentschik `config.InstanceConfig.OcPath` is set in start.go. So unless a profile is started config.InstanceConfig.OcPath is not set.  When we are trying to remove the context , it means there is fair chance that the old profile will not have the oc binary cached and `config.InstanceConfig.OcPath` would be empty.